src/Spelling/WordParser.cs

```diff
@@ -0,0 +1,566 @@
+    /// </summary>
+    public class WordParser
+    {
+        // Fields
+        private const char NullChar = '\0';
+
+        private readonly StringBuilder _buffer;
+
+        private readonly string _text;
+
+        private readonly WordParserOptions _wordParserOptions;
+
+        /// <summary>
+        /// The recognized words
+        /// </summary>
+        private readonly ICollection<string> _recognizedWords;
+
+        private int _index;
+
+        private char _prefixChar;
+
+        private string _peekedWord;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WordParser"/> class.
+        /// </summary>
+        /// <param name="text">
+        /// The text.
+        /// </param>
+        /// <param name="options">
+        /// The options.
+        /// </param>
+        /// <param name="recognizedWords">
+        /// The recognized words.
+        /// </param>
+        public WordParser(string text, WordParserOptions options, ICollection<string> recognizedWords = null)
+            : this(text, options, recognizedWords, NullChar)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WordParser"/> class.
+        /// </summary>
+        /// <param name="text">The text.</param>
+        /// <param name="options">The options.</param>
+        /// <param name="recognizedWords">The recognized words.</param>
+        /// <param name="prefixChar">The prefix char.</param>
+        /// <exception cref="System.ArgumentNullException">text is null</exception>
+        /// <exception cref="System.ComponentModel.InvalidEnumArgumentException">options is invalid</exception>
+        internal WordParser(string text, WordParserOptions options, ICollection<string> recognizedWords, char prefixChar)
+        {
+            if (text == null)
+            {
+                throw new ArgumentNullException(nameof(text));
+            }
+
+            if ((options < WordParserOptions.None) || (options > (WordParserOptions.SplitCompoundWords | WordParserOptions.IgnoreMnemonicsIndicators)))
+            {
+                throw new InvalidEnumArgumentException(nameof(options), (int)options, typeof(WordParserOptions));
+            }
+
+            _text = text;
+            _wordParserOptions = options;
+            _buffer = new StringBuilder(text.Length);
+            _recognizedWords = recognizedWords ?? new Collection<string>();
+            _prefixChar = prefixChar;
+        }
+
+        private bool SkipMnemonics => (_wordParserOptions & WordParserOptions.IgnoreMnemonicsIndicators) == WordParserOptions.IgnoreMnemonicsIndicators;
+
+        private bool SplitCompoundWords => (_wordParserOptions & WordParserOptions.SplitCompoundWords) == WordParserOptions.SplitCompoundWords;
+
+        /// <summary>
+        /// The next word.
+        /// </summary>
+        /// <returns>
+        /// The System.String.
+        /// </returns>
+        public string NextWord()
```

Comment:
❕ Needs to be rewritten to return a non-allocating view of the original string, or it will not scale

Reply:
No longer applies